neps/nep-0518.md

```diff
@@ -0,0 +1,337 @@
+- The Transaction Relayer validates the Ethereum transaction, converts it into a NEAR-compatible format, and sends it to Bob’s Wallet Contract at `0xb794f5ea0ba39494ce839613fffba74279579268`.
+- The **Wallet Contract** decodes the Ethereum transaction, verifies the signature against the Ethereum address, and executes the corresponding action on the DefiEx smart contract.
+
+This method meets the expectations of both NEAR developers and Ethereum users.
+
+#### [COMPLEX] Example Transaction Flow: NEP-141 Transfer from Metamask
+
+- Charlie, a Metamask user, controls the Ethereum address `0xb794f5ea0ba39494ce839613fffba74279579268`.
+- Declan, another Metamask user, controls the Ethereum address `0xf977814e90da44bfa03b6295a0616a897441acec`.
+- Charlie, having acquired USDT on a centralized exchange, wishes to transfer 10 NEAR-native USDT to Declan’s address.
+- Charlie adds NEAR-native USDT to Metamask, using an address derived from the `keccak-256` hash of the NEAR account `usdt.tether-token.near`.
+- Metamask, via RPC Translator, displays USDT as a standard ERC20 token.
+- Charlie initiates a transfer from Metamask to Declan. This is processed as a standard ERC20 `transfer` method call.
+- The Transaction Relayer converts this into a NEAR transaction and sends it to the NEAR Blockchain.
+- The Wallet Contract identifies this as a special ERC-20 transfer, queries the **Ethereum Translation Contract** for a corresponding NEAR contract, and constructs an action for the `usdt.tether-token.near` contract.
+  - First, it checks if Declan’s account is registered with USDT, funding the storage if necessary.
+  - Next, it processes the `ft_transfer` action, attaching `1 yoctoNEAR` as per [NEP-141](https://nomicon.io/Standards/Tokens/FungibleToken/Core).
+- Post-transaction, the RPC Translator provides ERC-20 transaction logs to Metamask, confirming the transfer to Charlie.
+
+This transaction flow adheres to the expectations of Ethereum users.
+
+### Wallet Selector and Transaction Transformation
+
+Ethereum transactions, [RLP encoded](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/), encapsulate the following information:
+
+- `To` address of the transaction,
+- `From` address (derived from the signature),
+- `Gas price`,
+- `Gas limit`,
+- `Nonce`,
+- Base token transfer amount within a transaction (`value`),
+- `Chain ID`,
+- `Data field`,
+- `Signature` (yielding the public key and `From` address).
+
+For function calls to a smart contract, the data field contains:
+
+- The `Function Selector`,
+- Function call parameters.
+
+Both `To` and `From` addresses, encoded in hexadecimal and prefixed with “0x”, comprise 20 bytes. The `From` address is the right-most 20 bytes of the **Keccak-256** hash of the binary public key of the transaction sender (an **EOA**, externally owned address). The To address signifies either the recipient EOA or an on-chain contract address.
```

Comment:
Side note: I don't see https://github.com/near/NEPs/issues/518#issuecomment-1853942353 having been addressed in the NEP (collection of NEPs should be in some ways as self-contained as feasible; keeping motivation as a comment somewhere is not discoverable to most readers).